src/libraries/System.Diagnostics.DiagnosticSource/tests/ActivityTests.cs

```diff
@@ -263,33 +263,34 @@ public void IdGenerationInternalParent()
             //start 2 children in different execution contexts
             Task.Run(() => child1.Start()).Wait();
             Task.Run(() => child2.Start()).Wait();
-#if DEBUG
-            Assert.Equal($"|{parent.RootId}.{child1.OperationName}-1.", child1.Id);
-            Assert.Equal($"|{parent.RootId}.{child2.OperationName}-2.", child2.Id);
-#else
-            Assert.Equal($"|{parent.RootId}.1.", child1.Id);
-            Assert.Equal($"|{parent.RootId}.2.", child2.Id);
-#endif
+
```

Comment:
Do you understand why are the IDs different between debug and release? I am not able to tell whether this change is right without understanding it.

Once you figure out why the IDs are different between debug and release, it would be nice to add it as a comment.

Reply:
Ok, I'll debug it and see why we get different IDs and then add a comment.